ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoverySuspendTest.java

```diff
@@ -0,0 +1,205 @@
+        _jtaEnvironmentBean.setXaResourceOrphanFilters(List.of(xaResourceOrphanFilter));
+        // the test set of modules
+        _recoveryConfig.setRecoveryModuleClassNames(Arrays.asList(modules));
+
+        _jtaEnvironmentBean.setXaResourceRecoveryClassNames(
+                List.of(BytemanControlledXAResourceRecovery.class.getName()));
+
+        // obtain a new RecoveryManager with the above config:
+        _manager = RecoveryManager.manager(RecoveryManager.INDIRECT_MANAGEMENT);
+        // recovery can start
+        _manager.startRecoveryManagerThread();
+    }
+
+    @BeforeEach
+    public void beforeEach() {
+        // In case the Recovery Manager was suspended, this will resume it
+        _manager.resume();
+        // Let's clean
+        ThreadActionData.purgeActions();
+    }
+
+    @AfterEach
+    public void afterEach() {
+        if (!TxControl.isEnabled()) {
+            // Re-activate transaction creation
+            TxControl.enable();
+        }
+    }
+
+    @Test
+    @BMScripts(scripts = {
+            @BMScript("RecoverySuspendTest/recoverySuspendTest_BytemanControlledXAResource"),
+            @BMScript("RecoverySuspendTest/recoverySuspendTest_PeriodicRecovery")
+    })
+    public void testSuspensionWhenThereArentTransactionsToRecover() {
+
+        // Make sure that the test environment is ready
+        BytemanControlledXAResource.resetCommitCallCounter();
+        BytemanControlledXAResource.resetRollbackCallCounter();
+        // The following instruction makes the BytemanControlledXAResource commit without failures.
+        BytemanControlledXAResource.setGreenFlag();
+        BytemanControlledXAResource.setCommitReturn(XAResource.XA_OK);
+
+        // BytemanControlledXAResource.getCommitCallCounter() should be 1 as:
+        // - One invocation from the normal commit procedure should pass
+        startTest(1,0, 0);
+    }
+
+    @Test
+    @BMScripts(scripts = {
+            @BMScript("RecoverySuspendTest/recoverySuspendTest_BytemanControlledXAResource"),
+            @BMScript("RecoverySuspendTest/recoverySuspendTest_FailTest")
+    })
+    public void testSuspensionWhenThereIsXATxnToRecover() {
+
+        // Make sure that the test environment is ready
+        BytemanControlledXAResource.resetCommitCallCounter();
+        BytemanControlledXAResource.resetRollbackCallCounter();
+        BytemanControlledXAResource.resetGreenFlag();
+        BytemanControlledXAResource.setCommitReturn(XAException.XAER_RMFAIL);
+
+        // BytemanControlledRecord.getCommitCallCounter() should be 3 as:
+        // - One invocation from the normal commit procedure should fail
+        // - One invocation from the recovery process should fail
+        // - One invocation from the recovery process should pass
+        startTest(1, 2, 0);
+    }
+
+    private void startTest(int numberOfCommits, int numberOfRollbacks, int timeout) {
+
+        createXATransaction(new BytemanControlledXAResource(), timeout);
```

Comment:
Can we verify after this call that no rollbacks where issued yet? It might be a bit unpredictable though as the PeriodicRecovery might have done it...

Reply:
I have reworked this part of the class. WDYT?

Reply:
I am not spotting it actually.  Apart from the unpredictability I would imagine something like: 
`Assertions.assertEquals(0, BytemanControlledXAResource.getRollbackCallCounter());`

Reply:
What I did is:
* `_jtaEnvironmentBean.setOrphanSafetyInterval(periodicRecoveryPeriod);`, which would make the amount of time needed for a transaction to be considered by XARecoveryModule longer
* Start the recovery manager just before the transaction gets created

With these modifications, I don't think we need to `Assertions.assertEquals(0, BytemanControlledXAResource.getRollbackCallCounter());`. WDYT?

Reply:
Ah, OK. Personally I think I would put the assertion in anyway (in case someone has it paused in the debugger and doesn't realise and gets confused) but the gist of what you are saying +1.

It's not mandatory from my point of view to make the change I mentioned, just a nice to have I think